libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/MapStyleLoader.kt

```diff
@@ -0,0 +1,88 @@
+package com.mapbox.navigation.dropin
+
+import android.content.Context
+import android.content.res.Configuration
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.maps.MapboxMap
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.observable.eventdata.StyleLoadedEventData
+import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+
+/**
+ * Lifecycle aware class that:
+ *  - loads initial map [Style] in onCreate
+ *  - reloads map [Style] on either [NavigationViewOptions.mapStyleUriNight] or
+ *   [NavigationViewOptions.mapStyleUriDay] flow changes
+ *  - registers itself as OnStyleLoadedListener and updates [loadedMapStyle] StateFlow value
+ *   every time new [Style] is fully loaded
+ */
+internal class MapStyleLoader(
+    private val context: Context,
+    private val mapboxMap: MapboxMap,
+    private val options: NavigationViewOptions,
+    private val loadedMapStyle: MutableStateFlow<Style?>,
+) : OnStyleLoadedListener, DefaultLifecycleObserver {
```

Comment:
```suggestion
) : DefaultLifecycleObserver {
```

Small opinion here. Move the callback interface inside, doesn't need to be exposed on the top level

Reply:
Done